### PR TITLE
Migrate to Spine 0.10.76 and base 0.10.63

### PR DESCRIPTION
**Pull request by vlad-lubenskyi**
This PR:

- bumps the version to `0.10.76`;
- bumps Spine's version to `0.10.76`;
- bumps `base` to `0.10.63`;
- migrates to Java 8 API;
  - replaces Guava's functions, predicates, and optionals with the native alternatives;
  - uses `Streams` for filtering and mapping collections.
- reflects changes made in Spine since the last update:
  - updates Datastore indexes (both in `index.yaml` and on `spine-dev`);
  - fixes imports to the moved classes;
  - `BoundedContextBuilder` vs `BoundedContext.Builder`.

### CODE REVIEW

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

File: datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java [hunk 1]
```diff
@@ -169,20 +168,42 @@ static Filter activeEntity() {
      */
     private static Collection<Filter> toFilters(Collection<CompositeQueryParameter> parameters,
                                                 ColumnFilterAdapter columnFilterAdapter) {
-        FluentIterable<CompositeQueryParameter> params = from(parameters);
-        FluentIterable<CompositeQueryParameter> conjunctionParams =
-                params.filter(isConjunctive);
-        FluentIterable<CompositeQueryParameter> disjunctionParams =
-                params.filter(isDisjunctive);
-        Optional<CompositeQueryParameter> firstParam = conjunctionParams.first();
+        List<CompositeQueryParameter> conjunctionParams = parameters.stream()
```
Comment (armiol) [inline datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java]: Please extract `parameters.stream()` into a variable.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java [hunk 2]
```diff
@@ -169,20 +168,42 @@ static Filter activeEntity() {
...
     }
 
+    /**
+     * Merges conjunctive parameters into a single {@link Filter}.
+     *
+     * @param conjunctiveParams list of parameters
+     * @return resulting filter or {@code Optional.empty()} if there {@code conjunctiveParams} is
+     * empty
+     */
+    private static Optional<CompositeQueryParameter> mergeConjunctiveParameters(
```
Comment (armiol) [inline datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java:193]: I don't understand why this is in `gcloud-java`, while this should most likely be in `core-java`.

Let's discuss in person.

**Review by armiol — CHANGES_REQUESTED** [review]
@vlad-lubenskyi LGTM in general. A few points, including one to discuss.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java (hunk 2 above)
Comment (armiol) [reply datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java:193]: As discussed, this should be reflected as an issue. LGTM for now.

**Review by armiol — APPROVED** [review]
@vlad-lubenskyi LGTM after you extract a variable (please see the only remaining comment).

**Pull request merged**